fuel-vm/src/checked_transaction.rs

```diff
@@ -186,40 +191,91 @@ pub trait IntoChecked: FormatValidityChecks + Sized {
     fn into_checked(
         self,
         block_height: BlockHeight,
-        params: &ConsensusParameters,
-        gas_costs: &GasCosts,
+        consensus_params: ConsensusParams<'_>,
```

Comment:
Do you want to move `GasCosts` to `ConsensusParams` in a separate PR? Because it was an initial reason for this PR(have one type that includes all data required for consensus verification)=D

Reply:
Okay. So, `GasCosts` is included on the `TransactionBuilderExt`:
```
fn finalize_checked(
        &mut self,
        height: BlockHeight,
        gas_costs: GasCosts,
    ) -> Checked<Tx>;
```

Which implies that, even though the transaction builder might hold the consensus params, it might not hold the `GasCosts`. 

That might just be an artifact and it can hold the `GasCosts` too.

Reply:
I'll try that. 

I'm not really sure why the `TransactionBuilderExt` trait exists at all. I don't see anything that depends on it. I wonder why it can't just be a method on the `TransactionBuilder` directly.